src/main/test/surfrdUtils_test/test_surfrdUtils.pf

```diff
@@ -43,6 +43,195 @@ contains
+     wt_lunit(begg:,isturb_md) = (/ input_value(8),  input_value(1), input_value(3)/)
+     wt_lunit(:,:) = wt_lunit(:,:) / 100._r8
+     call check_sums_equal_1( wt_lunit, begg, "test_check_sums_add_to_1", &
+                              "should not trigger an error")
+
+     do test = 1, tests
+
+        ! OUTPUT VALUES EXPECTED
+        if (test == test_toosmall_is_0) then
+           wt_expected = wt_lunit
+        else if (test == test_toosmall_is_1) then
+           ! Keeping landunits exceeding 1% of the grid cell, hence
+           ! we sum the first 6 entries of input_value
+           wt_expected = 100._r8 * wt_lunit / sum(input_value(1:6))
+           wt_expected(begg, isturb_hd) = 0._r8
+           wt_expected(begg+1, istcrop) = 0._r8
+           wt_expected(begg+2, istdlak) = 0._r8
+        else if (test == test_toosmall_is_8) then
+           ! Keeping landunits exceeding 8% of the grid cell, hence
+           ! we sum the first 3 entries of input_value
+           wt_expected = 100._r8 * wt_lunit / sum(input_value(1:3))
+           wt_expected(begg, istdlak:isturb_hd) = 0._r8
+           wt_expected(begg+1, istcrop:istwet) = 0._r8
+           wt_expected(begg+2, istsoil:istdlak) = 0._r8
+        else if (test == test_toosmall_is_39 .or. test == test_toosmall_is_100) then
+           ! Keeping the largest landunit, hence
+           ! we sum the first entry of input_value
+           wt_expected = 100._r8 * wt_lunit / input_value(1)
+           wt_expected(begg, istcrop:isturb_hd) = 0._r8
+           wt_expected(begg+1, istcrop:isturb_hd) = 0._r8
+           wt_expected(begg+2, istsoil:istdlak) = 0._r8
+           wt_expected(begg+2, isturb_hd:isturb_md) = 0._r8
+        end if
+
+        call check_sums_equal_1( wt_expected, begg, "test_check_sums_add_to_1", &
+                                 "should not trigger an error for wt_expected")
+
+        ! Collapse landunits
+        wt_in_out = wt_lunit  ! reset argument for next call
+        call collapse_individual_lunits(wt_in_out, begg, endg, &
+                                        toosmall_soil(test), &
+                                        toosmall_crop(test), &
+                                        toosmall_glacier(test), &
+                                        toosmall_lake(test), &
+                                        toosmall_wetland(test), &
+                                        toosmall_urban(test))
+
+        ! Now check that are correct
+        call check_sums_equal_1( wt_in_out, begg, "test_check_sums_add_to_1", &
+                                 "should not trigger an error for wt_in_out")
+
+        @assertEqual(wt_in_out(begg:,:), wt_expected(begg:,:), tolerance=tol)
+
+     end do
+
+     deallocate( wt_expected )
+     deallocate( wt_in_out )
+     deallocate( wt_lunit )
+
+     call pftcon%clean()
+
+  end subroutine test_collapse_individual_lunits
+
+  @Test
+  subroutine test_collapse_urban()
```

Comment:
Add another test subroutine, that tests using the n_dom_* and the too_small_* all together at once.

Reply:
Done.